packages/graph/src/graph2/layouts/forceDirected.ts

```diff
@@ -39,6 +39,8 @@ export class ForceDirectedBase extends Layout {
 
             this._simulation = d3ForceSimulation(vertices.map(v => {
                 const { width, height } = v.element.node().getBBox();
+                v.fx = (this._options.pinCentroid && v.centroid) ? size.width / 2 : undefined;
+                v.fy = (this._options.pinCentroid && v.centroid) ? size.height / 2 : undefined;
```

Comment:
I could see this causing some confusion if the user attempts to drag the centroid.

Reply:
If a user drags the centroid during an animated spring, then it (currently) effectively unpins it when they release it (which could be improved upon).